test/jdk/java/lang/Class/getEnclosingClass/EnclosingClassTest.java

```diff
@@ -59,71 +48,153 @@
  * e o x x x
  */
 
-import java.util.List;
-import java.util.LinkedList;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
 import java.lang.reflect.Field;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
 import common.TestMe;
+import jdk.test.lib.JDKToolFinder;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+import jdk.test.lib.util.FileUtils;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/*
+* @test
+* @bug 4992173 4992170
+* @library /test/lib
+* @modules jdk.compiler
+* @build jdk.test.lib.process.* EnclosingClassTest
+* @run testng/othervm EnclosingClassTest
+* @summary Check getEnclosingClass and other methods
+* @author Peter von der Ah\u00e9
+*/
+
 
 public class EnclosingClassTest {
-    static void info(Class<?> c, Class<?> encClass, String desc) {
-        if (!"".equals(desc))
+    private static final String SRC_DIR = System.getProperty("test.src");
+    private static final String ENCLOSING_CLASS_SRC = SRC_DIR + "/EnclosingClass.java";
```

Comment:
```suggestion
    private static final Path ENCLOSING_CLASS_SRC = Paths.get(SRC_DIR, "EnclosingClass.java");
```

Use Path::toString to convert to a String where needed.